operator-common/src/main/java/io/strimzi/operator/common/operator/resource/ClusterRoleOperator.java

```diff
@@ -4,59 +4,87 @@
 import io.vertx.core.Future;
 import io.vertx.core.Vertx;
 
-/**
- * This class is a temporary work-around for the fact that Fabric8 doesn't
- * yet support an API for manipulating Kubernetes ClusterRoles
- * @deprecated This can be removed once support for ClusterRoles and ClusterRoleBindings is in Fabric8.
- */
-@Deprecated
-public class ClusterRoleOperator extends WorkaroundRbacOperator<ClusterRoleOperator.ClusterRole> {
-
-    public ClusterRoleOperator(Vertx vertx, KubernetesClient client) {
-        super(vertx, client, "rbac.authorization.k8s.io", "v1beta1", "clusterroles");
-    }
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
 
-    public static class ClusterRole {
-        private final String resource;
+public class ClusterRoleOperator extends AbstractNonNamespacedResourceOperator<KubernetesClient, KubernetesClusterRole, KubernetesClusterRoleList, DoneableKubernetesClusterRole, Resource<KubernetesClusterRole, DoneableKubernetesClusterRole>> {
 
-        public ClusterRole(String yaml) {
-            this.resource = convertYamlToJson(yaml);
-        }
+    /**
+     * Constructor
+     * @param vertx The Vertx instance
+     * @param client The Kubernetes client
+     */
 
-        private String convertYamlToJson(String yaml) {
-            try {
-                ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
-                Object obj = yamlReader.readValue(yaml, Object.class);
-                ObjectMapper jsonWriter = new ObjectMapper();
-                return jsonWriter.writeValueAsString(obj);
-            } catch (IOException e)   {
-                throw new RuntimeException(e);
-            }
-        }
-
-        public String toString() {
-            return resource;
-        }
+    public ClusterRoleOperator(Vertx vertx, KubernetesClient client) {
+        super(vertx, client, "ClusterRole");
     }
 
-    private String urlWithoutName() {
-        return baseUrl + "apis/" + group + "/" + apiVersion + "/" + plural;
+    @Override
+    protected MixedOperation<KubernetesClusterRole, KubernetesClusterRoleList, DoneableKubernetesClusterRole, Resource<KubernetesClusterRole, DoneableKubernetesClusterRole>> operation() {
+        return client.rbac().kubernetesClusterRoles();
     }
 
-    private String urlWithName(String name) {
-        return urlWithoutName() + "/" + name;
+    @Override
+    protected Future<ReconcileResult<KubernetesClusterRole>> internalPatch(String name, KubernetesClusterRole current, KubernetesClusterRole desired) {
+        return Future.succeededFuture(ReconcileResult.noop(current));
     }
 
-    public Future<Void> reconcile(String name, ClusterRole resource) {
-        return doReconcile(urlWithoutName(), urlWithName(name), resource);
-    }
+    public static KubernetesClusterRole convertYamlToClusterRole(String yaml) {
```

Comment:
I don't understand why you needed to write this method. It doesn't seem to be the best way of doing whatever is needed? In particular why can't you just use something like `yamlReader.readValue(yaml, KubernetesClusterRole.class)`?